### ISSUE TITLE
Produce sorted output

### ISSUE THREAD

**Issue by AMDmi3** [AMDmi3/omnilinter#33]
Output should be sorted. That given, there should be an option to return some matches as early as possible, as processing all repositories can be long.

Implementation of this feature will be totally different before and after #25, so it makes sense to postpone it not to do the work twice.

When this feature is implemented, however:
- tests should be modified to take order into account (currently they explicitly sort output to ignore ordering)
- care should be taken to sort paths properly, as sorting paths is different from sorting strings (paths should be ordered component-wise)

**Comment by AMDmi3** [comment]
As of now, the output is sorted, as we use sorting when walking roots, process files by line and apply rules in the order specified in the ruleset, so the order is `root->file->line->rule`. However, this may change if we process multiple roots in parallel or if we introduce more complex condition handling.

**Comment by AMDmi3** [comment]
Done in a1334ec290167a2411af479d0db487d67163c86d
